nameable.rb

```diff
@@ -0,0 +1,30 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
+end
+
+class Decorator < Nameable
+  def initialize(nameable)
+    @nameable = nameable
+    super()
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    super.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    return super[0, 10] if super.length > 10
+
+    super
+  end
+end
```

Comment:
- [ ] I will encourage you to extract every class to a different file which is the professional and requirement in this project.